entity-service-change-event-generator/src/main/java/org/hypertrace/entity/service/change/event/impl/EntityChangeEventGeneratorImpl.java

```diff
@@ -147,6 +168,33 @@ private void sendUpdateNotification(
     }
   }
 
+  private boolean shouldSendNotification(
+      RequestContext requestContext, Entity prevEntity, Entity currEntity) {
+    String entityType = prevEntity.getEntityType();
+    Entity.Builder prevEntityBuilder = prevEntity.toBuilder();
+    Entity.Builder currEntityBuilder = currEntity.toBuilder();
+    this.changeNotificationSkipAttributeList.forEach(
+        attributeId -> {
+          Optional<AttributeMetadata> attributeScopeKey =
+              this.entityAttributeMapping.getAttributeMetadataByAttributeId(
+                  requestContext, attributeId);
+          if (attributeScopeKey.isPresent()
```

Comment:
nit: prefer ifPresent rather than explicit isPresent + get's (produces the same result, but get() requires logic, while ifPresent will give you compile-time validation).